src/network/networkd-mdb.c

```diff
@@ -0,0 +1,247 @@
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Network *network = userdata;
+        _cleanup_(mdb_entry_free_or_set_invalidp) MdbEntry *mdb_entry = NULL;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = mdb_entry_new_static(network, filename, section_line, &mdb_entry);
+        if (r < 0)
+                return log_oom();
+
+        r = in_addr_from_string_auto(rvalue, &mdb_entry->family, &mdb_entry->group_addr);
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r, "Cannot parse group address: %m");
+                return 0;
+        }
+
+        if (!in_addr_is_multicast(mdb_entry->family, &mdb_entry->group_addr)) {
+                log_syntax(unit, LOG_WARNING, filename, line, 0, "Address %s is not a multicast group", rvalue);
+                return 0;
+        }
+
+        mdb_entry = NULL;
+
+        return 0;
+}
+
+/* remove and MDB entry. */
+MdbEntry *mdb_entry_free(MdbEntry *mdb_entry) {
+        if (!mdb_entry)
+                return NULL;
+
+        if (mdb_entry->network) {
+                LIST_REMOVE(static_mdb_entries, mdb_entry->network->static_mdb_entries, mdb_entry);
+                assert(mdb_entry->network->n_static_mdb_entries > 0);
+                mdb_entry->network->n_static_mdb_entries--;
+
+                if (mdb_entry->section)
+                        hashmap_remove(mdb_entry->network->mdb_entries_by_section, mdb_entry->section);
+        }
+
+        network_config_section_free(mdb_entry->section);
+
+        return mfree(mdb_entry);
+}
+
+static int set_mdb_handler(sd_netlink *rtnl, sd_netlink_message *m, Link *link) {
+        int r;
+
+        assert(link);
+
+        if (IN_SET(link->state, LINK_STATE_FAILED, LINK_STATE_LINGER))
+                return 1;
+
+        r = sd_netlink_message_get_errno(m);
+        if (r < 0 && r != -EEXIST) {
+                log_link_message_warning_errno(link, m, r, "Could not add MDB entry");
+                link_enter_failed(link);
+                return 1;
+        }
+
+        return 1;
+}
+
+/* send a request to the kernel to add an MDB entry */
+int mdb_entry_configure(Link *link, MdbEntry *mdb_entry) {
+        _cleanup_(sd_netlink_message_unrefp) sd_netlink_message *req = NULL;
+        struct br_mdb_entry entry = {};
+        int r;
+
+        assert(link);
+        assert(link->network);
+        assert(link->network->bridge);
+        assert(link->manager);
+        assert(mdb_entry);
+
+        entry.state = MDB_PERMANENT;
+        entry.ifindex = link->ifindex;
+        entry.vid = mdb_entry->vlan_id;
```

Comment:
use structured initialization for threse three here then, and drop if from the variable declaration above, i.e. 

```c
entry = (struct br_mdb_entry) {
        .state = MDB_PERMANENT,
        .ifindex = link->ifindex,
        .vid = mdb_entry->vlan_id,
};
```

And yeah, last time we analyzed this gcc generates better code when you do zero initialization as side effect of structured initialization instead of separately
```

Reply:
Alright. Done :)